WordPress/src/main/java/org/wordpress/android/ui/stats/refresh/utils/StatsNavigator.kt

```diff
@@ -0,0 +1,160 @@
+package org.wordpress.android.ui.stats.refresh.utils
+
+import android.content.Intent
+import android.support.v4.app.FragmentActivity
+import org.wordpress.android.R
+import org.wordpress.android.fluxc.model.SiteModel
+import org.wordpress.android.fluxc.network.utils.StatsGranularity
+import org.wordpress.android.fluxc.network.utils.StatsGranularity.DAYS
+import org.wordpress.android.fluxc.network.utils.StatsGranularity.MONTHS
+import org.wordpress.android.fluxc.network.utils.StatsGranularity.WEEKS
+import org.wordpress.android.fluxc.network.utils.StatsGranularity.YEARS
+import org.wordpress.android.ui.ActivityLauncher
+import org.wordpress.android.ui.WPWebViewActivity
+import org.wordpress.android.ui.stats.StatsTimeframe
+import org.wordpress.android.ui.stats.StatsUtils
+import org.wordpress.android.ui.stats.models.StatsPostModel
+import org.wordpress.android.ui.stats.refresh.NavigationTarget
+import org.wordpress.android.ui.stats.refresh.NavigationTarget.AddNewPost
+import org.wordpress.android.ui.stats.refresh.NavigationTarget.SharePost
+import org.wordpress.android.ui.stats.refresh.NavigationTarget.ViewAuthors
+import org.wordpress.android.ui.stats.refresh.NavigationTarget.ViewClicks
+import org.wordpress.android.ui.stats.refresh.NavigationTarget.ViewCommentsStats
+import org.wordpress.android.ui.stats.refresh.NavigationTarget.ViewCountries
+import org.wordpress.android.ui.stats.refresh.NavigationTarget.ViewFollowersStats
+import org.wordpress.android.ui.stats.refresh.NavigationTarget.ViewPost
+import org.wordpress.android.ui.stats.refresh.NavigationTarget.ViewPostDetailStats
+import org.wordpress.android.ui.stats.refresh.NavigationTarget.ViewPostsAndPages
+import org.wordpress.android.ui.stats.refresh.NavigationTarget.ViewPublicizeStats
+import org.wordpress.android.ui.stats.refresh.NavigationTarget.ViewReferrers
+import org.wordpress.android.ui.stats.refresh.NavigationTarget.ViewSearchTerms
+import org.wordpress.android.ui.stats.refresh.NavigationTarget.ViewTag
+import org.wordpress.android.ui.stats.refresh.NavigationTarget.ViewTagsAndCategoriesStats
+import org.wordpress.android.ui.stats.refresh.NavigationTarget.ViewUrl
+import org.wordpress.android.ui.stats.refresh.NavigationTarget.ViewVideoPlays
+import org.wordpress.android.util.ToastUtils
+
+class StatsNavigator(
```

Comment:
I think it would be better if this class were injected and the `navigate` method had `site` and `activity` as parameters. Especially since it's used in 2 fragments.